### PR TITLE
[tensorflow] | [build, test]TF 2.4 SMP 1.2.0

### PR DESCRIPTION
**Pull request by akhilmehra**
*Issue #, if available:*

## PR Checklist
- [ X] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ X] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ X] (If applicable) I've documented below the tests I've run on the DLC image
- [ X] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [X ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
This dockerfile contains the changes required to build SM Model Parallel for TF 2.4

*Tests run:*

*DLC image/dockerfile:*
Tensorflow 2.4 Cuda 11 GPU Python 3

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by saimidu — APPROVED** [review]

**Comment by saimidu** [comment]
Let's merge this once all tests pass.

File: tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu [hunk 1]
```diff
@@ -20,6 +20,7 @@ ENV LC_ALL=C.UTF-8
 ENV KMP_AFFINITY=granularity=fine,compact,1,0
 ENV KMP_BLOCKTIME=1
 ENV KMP_SETTINGS=0
+ENV MANUAL_BUILD=0
```
Comment (tejaschumbalkar) [inline tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu:23]: Remove unused variable `MANUAL_BUILD `

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py [hunk 2]
```diff
@@ -37,7 +38,7 @@ def test_smmodelparallel(sagemaker_session, instance_type, ecr_image, tmpdir, fr
     instance_type = "ml.p3.16xlarge"
     _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
     image_cuda_version = get_cuda_version_from_tag(ecr_image)
-    if Version(image_framework_version) != Version("2.3.1") or image_cuda_version != "cu110":
+    if Version(image_framework_version) < Version("2.3.1") or image_cuda_version != "cu110":
         pytest.skip("Model Parallelism only supports CUDA 11 on TensorFlow 2.3")
```
[the remark this answers is not in the document]
Comment (akhilmehra) [reply test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py]: Done.

File: test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py [hunk 3]
```diff
@@ -76,7 +75,7 @@ def test_smmodelparallel_multinode(sagemaker_session, instance_type, ecr_image,
...
```
[the remark this answers is not in the document]
Comment (akhilmehra) [reply test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smmodelparallel.py]: Done.

File: tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu (hunk 1 above)
Comment (akhilmehra) [reply tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu:23]: This variable is used internally during SMP build time

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**